### PR TITLE
Fix/fetch from pbl api

### PR DESCRIPTION
**Pull request by petermisik**
Changed PBSEvent model to use Codable
Changed PBSCrew model to use Codable
Altered DataController to absorb Sorite Mission number, takeoff time, and land time.
Created Extension to make our JSON prettyPrinted.
Updated PBSEventCard and ImportEvents to use the new models.

### CODE REVIEW

**Review by jbethanco — CHANGES_REQUESTED** [review]
Not looking ready for primetime. Tried to add an event and got a crash here: 
image

A lot of commented out code. The JSON.swift file and the old data models all need to be removed. There seems to be some things that are observed objects/published vars that don't need to be.

File: PBLogging/Network/PBSModels.swift [hunk 1]
```diff
@@ -113,147 +117,69 @@ class PBSOrganization: ObservableObject, Identifiable {
...
+                        for event in results {
+                            self.events.append(event)
+                            print("I Added \((event.name)!)")
                         }
-                    }
+                } catch {
+                    print(error.localizedDescription)
                 }
-
+//                print("JSON: \(data.prettyPrintedJSONString!)")
```
[the remark this answers is not in the document]
Comment (petermisik) [reply PBLogging/Network/PBSModels.swift]: A lot of clean up between Sam and I.  Refactored and removed unused code.

File: PBLogging/Network/ImportEventsView.swift [hunk 2]
```diff
@@ -114,85 +142,80 @@ struct ImportEventsView: View {
...
 struct PBSOrganizationNavLink: View {
+    @EnvironmentObject var dataController: DataController
     @Binding var isShown: Bool
-    @ObservedObject var org: PBSOrganization
+    var organizationName: String
+    var events: [PBSEvent]
 
     var body: some View {
-        NavigationLink(destination: PBSEventsView(organization: org, isShown: $isShown)) {
+        NavigationLink(destination: PBSEventsView(isShown: $isShown, events: events).environmentObject(dataController)) {
```
Comment (jbethanco) [inline PBLogging/Network/ImportEventsView.swift:183]: .environmentObject modifier should be on next line.

File: PBLogging/Network/PBSModels.swift [hunk 3]
```diff
@@ -6,254 +6,131 @@
...
-
-        let crewLine2crew3 = PBSCrew()
-        crewLine2crew3.lastName = "Louie"
-
-        event2.crewLines = [crewLine2crew1, crewLine2crew2, crewLine2crew3]
-
-        pbsOrg.events = [event1, event2]
-
-        pbsUser.organizations.append(pbsOrg)
+    /// The following is for Swift UI previews.
```
Comment (jbethanco) [inline PBLogging/Network/PBSModels.swift:14]: If for previews, maybe just include in the preview struct.

File: PBLogging/Network/SampleJSONData.swift [hunk 4]
```diff
@@ -0,0 +1,885 @@
...
+//  SampleJSONData.swift
+//  Logging
+//
+//  Created by Pete Rogers on 6/15/21.
+//
+
+import Foundation
+
+// swiftlint:disable:next type_body_length
+class SampleJSONData {
```
Comment (jbethanco) [inline PBLogging/Network/SampleJSONData.swift:11]: Does this need to be a class? If we don't need inheritance or reference types, no need for a class. This looks like this should be an enum.

File: PBLogging/Network/SampleJSONData.swift [hunk 5]
```diff
@@ -0,0 +1,885 @@
+//
+//  SampleJSONData.swift
+//  Logging
+//
+//  Created by Pete Rogers on 6/15/21.
```
Comment (jbethanco) [inline PBLogging/Network/SampleJSONData.swift:5]: Who is Pete Rogers?

File: PBLogging/Network/ImportEventsView.swift [hunk 6]
```diff
@@ -222,8 +245,8 @@ struct CrewLineView: View {
         ImportEventsView(isShown: .constant(true), dataModel: PBSUser.sampleUserOneOrg)
     }
  }
-#else
-#endif
+ #else
+ #endif
 
 struct CardModifier: ViewModifier {
```
Comment (jbethanco) [inline PBLogging/Network/ImportEventsView.swift:251]: Unused modifier

File: PBLogging/Data/DataController.swift [hunk 7]
```diff
@@ -87,15 +87,20 @@ class DataController: ObservableObject {
...
-            crewLine.flyingOrganization = crewMember.extractedOrg
-            crewLine.person.lastName = crewMember.lastName
-            crewLine.person.pbsUUID = UUID(uuidString: crewMember.pbsID)
+            crewLine.flyingOrganization = crewMember.primaryOrganizationName ?? ""
+            crewLine.person.lastName = crewMember.lastName ?? ""
+            crewLine.person.pbsUUID = UUID(uuidString: crewMember.pbsID ?? "")
             crewLine.sortie = sortie
-            
+
             print("GREAT SUCCESS >>> \(crewLine)")
```
Comment (jbethanco) [inline PBLogging/Data/DataController.swift:104]: I'm bad about forgetting to remove these. Probably good to delete or if must haves for some reason, give them some context as to what is happening.

File: PBLogging/Network/PBSModels.swift [hunk 8]
```diff
@@ -6,254 +6,131 @@
...
+        
+        var pbsUser = PBSUser(info: info)
+        let dataContainer = SampleJSONData()
+        let jsonDecoder = JSONDecoder()
+        jsonDecoder.dateDecodingStrategy = .iso8601
+        do {
+            pbsUser = try jsonDecoder.decode(PBSUser.self, from: dataContainer.jsonUser1Org.data(using: .utf8)!)
+        } catch {
+            print("Decoding error")
+        }
```
Comment (jbethanco) [inline PBLogging/Network/PBSModels.swift:29]: This is a lot of supporting code for the previews as well as the JSON sample data. When I run the previews, they don't work. Probably should remove the supporting code and sample JSON data if the previews do not work. Add back when they do?

**Review by jbethanco — APPROVED** [review]
It is working, but may still need some cleanup.

File: PBLogging/Network/ImportEventsView.swift [hunk 9]
```diff
@@ -12,56 +12,61 @@ enum LoginState {
...
+   
 struct ImportEventsView: View {
-    @State var orgsProcessed = 0
-    @Binding var isShown: Bool
-    @State var isActivityIndicatorAnimating = false
-    @State var loginState: LoginState = .loggedIn
     @EnvironmentObject var network: NetworkClient
+    @Binding var isShown: Bool
+    @State var loginState: LoginState = .notLoggedIn
     @State var dataModel: PBSUser?
```
Comment (jbethanco) [inline PBLogging/Network/ImportEventsView.swift:20]: if the datamodel is a PBSUser, might be worth calling it pbsUser for clarity.

File: PBLogging/Network/ImportEventsView.swift [hunk 10]
```diff
@@ -12,56 +12,61 @@ enum LoginState {
...
-
             case .notLoggedIn:
                 LoginWebView(loginStatus: $loginState)
                     .transition(.move(edge: .bottom))
             }
         }
     }
     
-    func getEvents() {
+    private func checkDataModel(dataModel: PBSUser?, orgs: [String: String]?, events: [PBSEvent]?) {
```
Comment (jbethanco) [inline PBLogging/Network/ImportEventsView.swift:44]: Function name could be more clear, it's checking the dataModel which is a PBSUser, checking the orgs, and events and then downloading them if they are nil. "checkDataModel(...) at the call site does not inform the reader what is actually happening.

File: PBLogging/Network/PBSModels.swift [hunk 11]
```diff
@@ -6,254 +6,131 @@
...
-
-        event2.crewLines = [crewLine2crew1, crewLine2crew2, crewLine2crew3]
-
-        pbsOrg.events = [event1, event2]
-
-        pbsUser.organizations.append(pbsOrg)
+    /// The following is for Swift UI previews.
+       #if DEBUG
+       
+       static let sampleUserOneOrg: PBSUser = {
```
Comment (jbethanco) [inline PBLogging/Network/PBSModels.swift:17]: Indentation of this file is off.

File: PBLogging/Network/SampleJSONData.swift (hunk 5 above)
Comment (petermisik) [reply PBLogging/Network/SampleJSONData.swift:5]: Got to love blindly replacing a word.

File: PBLogging/Network/ImportEventsView.swift (hunk 6 above)
Comment (petermisik) [reply PBLogging/Network/ImportEventsView.swift:251]: Removed.

File: PBLogging/Network/PBSModels.swift (hunk 3 above)
Comment (petermisik) [reply PBLogging/Network/PBSModels.swift:14]: Commented out previews with a TODO.  Will resolve in the next sprint.

**Pull request merged**